uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader/Steps/1004.php

```diff
@@ -40,11 +48,34 @@ protected function getExistingNames($newActivityTypes) {
    * @param $activityType
    */
   protected function createActivityType($activityType) {
-    civicrm_api3('OptionValue', 'create', [
+    $params = [
       'option_group_id' => 'activity_type',
       'name' => $activityType,
-      'component_id' => 'CiviTask',
-    ]);
+    ];
+
+    if ($this->tasksEnabled) {
+      $params['component_id'] = 'CiviTask';
+    }
+
+    civicrm_api3('OptionValue', 'create', $params);
```

Comment:
Yeah, ideally we would like to create it as a CiviTask since it would then be usable in T&A, but since this is civihr we can't be sure T&A will be installed. If not, we still need the new activity types.